### PR TITLE
[HE] setup translation

### PR DESCRIPTION
**Pull request by davidpodless**
I've added what you asked, and now the PR is to main

1. was fixed.
2. we still have the problem of RTL/LTR:
The numbers are all RTL :-/
image

### CODE REVIEW

**Comment by DarthGandalf** [comment]
> luaotfload | db : Reload initiated (formats: otf,ttf,ttc); reason: Font "Times New Roman" not found.

Would the Liberation font we found for cyrillics work for Hebrew as well? See main_ru.tex how the font is setup

**Comment by davidpodless** [comment]
> > luaotfload | db : Reload initiated (formats: otf,ttf,ttc); reason: Font "Times New Roman" not found.
> 
> Would the Liberation font we found for cyrillics work for Hebrew as well? See main_ru.tex how the font is setup

It didn't work on my machine. 
I'll find something, is there a way to run the tests before sending PRs?

**Comment by DarthGandalf** [comment]
> It didn't work on my machine.

Sad, I'll try to check it locally. Well, anyway...

> is there a way to run the tests before sending PRs?

Don't worry about failing tests in PRs - that's exactly what they are for!

But if you wish, just run build locally - the difference with CI is that you have "times new roman" available on your machine, while CI doesn't have it, whereas the font we use for Cyrillic is checked in to assets/ directory here, and referenced in main_ru.tex by path.

If you put the font files there too, it should work fine - please mind the font's license though, I don't think Times New Roman itself is free and can be used like that

**Comment by davidpodless** [comment]
> > It didn't work on my machine.
> 
> Sad, I'll try to check it locally. Well, anyway...
> 
> > is there a way to run the tests before sending PRs?
> 
> Don't worry about sending PRs before fixing all the tests - that's exactly what tests in CI are for!
> 
> But if you wish, just run build locally - the difference with CI is that you have "times new roman" available on your machine, while CI doesn't have it, whereas the font we use for Cyrillic is checked in to assets/ directory here, and referenced in main_ru.tex by path.
> 
> If you put the font files there too, it should work fine - please mind the font's license though, I don't think Times New Roman itself is free and can be used like that

at least there is a way to update the same PR?

**Comment by qwrtln** [comment]
> at least there is a way to update the same PR?

Yes, just push to the branch you're working on, the PR will be updated.

**Comment by qwrtln** [comment]
Well, without Times New Roman it builds 🎉 Now let's try to fix those RTL and graphics problems.

**Comment by qwrtln** [comment]
@Heegu-sama, we need you to mark Hebrew checks as required.

**Comment by qwrtln** [comment]
Normally, in the setup PR, we take care of the title page and everything until table of contents. Without them working as intended, the document looks broken.

**Comment by DarthGandalf** [comment]
I've managed to fix numbers:

Выделение_584

and added a way to fix graphics, but it needs to be considered one by one.

See the diff at 3641753dc077b3209eb1868aee57abfc4dfa0b3c

TODO: tables became gray, probably they need similar treatment with wrapping something with explicit English

**Comment by DarthGandalf** [comment]
Hm, are the numbers supposed to be 123, or Hebrew characters? If Hebrew characters, some images will have to be updated - the green dots with digits 123 are baked in there

**Comment by DarthGandalf** [comment]
One more issue: in printable build the page number is in wrong place:

Выделение_585

**Comment by davidpodless** [comment]
> I've managed to fix numbers:
> 
> Выделение_584
> 
> and added a way to fix graphics, but it needs to be considered one by one.
> 
> See the diff at [3641753](https://github.com/Heegu-sama/Homm3BG/commit/3641753dc077b3209eb1868aee57abfc4dfa0b3c)
> 
> TODO: tables became gray, probably they need similar treatment with wrapping something with explicit English

Wow, it is amazing, you even managed to use the hebrew calendar. I love it :-)

**Comment by davidpodless** [comment]
> Hm, are the numbers supposed to be 123, or Hebrew characters? If Hebrew characters, some images will have to be updated - the green dots with digits 123 are baked in there

123 is what most people are using. 
we can use Hebrew letters, similar to the 4a 4b etc
using only letter is more traditional way for religious texts (like the bible(

**Comment by davidpodless** [comment]
> Normally, in the setup PR, we take care of the title page and everything until table of contents. Without them working as intended, the document looks broken.

I agree, but working on fixing it isn't related to the translation itself, so I don't want it to be a gate for the translation

**Comment by davidpodless** [comment]
With a help from GPT, I've managed to make the page numbers to looks as they should

image

the numbers in the text aren't fixed, but it could be fixed locally with the translate itself, so it isn't much of an issue

last page picture looks off
image

**Comment by DarthGandalf** [comment]
I wrapped the first two pictures with `\textenglish{...}`, some more pictures (like that last page) needs that too.

If you wrap the setup image with it, it puts the green dots on correct places again.

**Comment by DarthGandalf** [comment]
The griffin on page 16 may need to be flipped instead... But we'll get to that when working on those chapters

**Comment by DarthGandalf** [comment]
afc52109cb9239991dd0e55c4beecddd886e99ed partially fixes printable

File: metadata.tex
Comment (DarthGandalf) [inline metadata.tex:178]: I think you've missed `}` somewhere below

Comment (DarthGandalf) [reply metadata.tex:178]: nvm, found it

**Comment by davidpodless** [comment]
I fixed the cover picture, flipped the griffin in page 16 and the dots in page 9
image

image

image



everything else could be fixed the same way in future PRs

**Comment by davidpodless** [comment]
I have no idea why check_po4a failed. I'll appreciate any help in this regard

**Comment by DarthGandalf** [comment]
@qwrtln how do you think, the `\textenglish` wrapping should be done in the original, like what I prototyped for title page, or purely in the translation, like what's here for the back cover and setup image?

**Comment by DarthGandalf** [comment]
@davidpodless 

>  There are uncommited changes to translations:
 M translations/back_cover.tex/he.po
 M translations/setup.tex/he.po
 M translations/title_page.tex/cs.po
 M translations/title_page.tex/de.po
 M translations/title_page.tex/es.po
 M translations/title_page.tex/fr.po
 M translations/title_page.tex/he.po
 M translations/title_page.tex/pl.po
 M translations/title_page.tex/ru.po
 M translations/title_page.tex/title_page.tex.pot
 M translations/title_page.tex/ua.po

This check checks that all translation files have the same original strings as the original .tex files, and that they are formatted properly.

title_page is in the list, because I modified original title_page.tex, but didn't run po4a to update the translations. back_cover.tex and setup.tex simply needs a run of `po4a po4a.cfg`, it will reformat it automatically. It can be run either from command line, or using a special bot in this repo, if someone comments "po4a check" or "po4a commit"

**Comment by davidpodless** [comment]
OK, now we are done 😄

File: translations/title_page.tex/cs.po
Comment (tomaas-zeman) [inline translations/title_page.tex/cs.po:38]: Delete the fuzzy part

```suggestion
```

File: translations/title_page.tex/cs.po
Comment (tomaas-zeman) [inline translations/title_page.tex/cs.po:49]: Update this with `msgid` text and don't forget to replace

```
\\layout/cover_subtitle.png
```

with 

```
translations/assets/cs/cover_subtitle.png
```

**Review by tomaas-zeman — CHANGES_REQUESTED** [review]

**Review by tomaas-zeman — APPROVED** [review]

**Comment by davidpodless** [comment]
Instead of making noises to all.
I moved the title page chagnes into hebrew

@DarthGandalf I think that answer your question. 
The only downside I see for it to be inside hebrew is if someone will decide to do an Arabic translation as well (or any other RTL language as Persian, Urdu etc) but they will be able to use the same method we are using for hebrew

**Comment by davidpodless** [comment]
OK, NOW we are ready 😄 

I would like to merge this if possible

@DarthGandalf @qwrtln @Heegu-sama

**Comment by DarthGandalf** [comment]
> @DarthGandalf I think that answer your question.

I see certain benefits both ways, but okay.

Tables still look broken, but that can be fixed when the translation progress reaches one.

**Review by DarthGandalf — APPROVED** [review]

**Comment by davidpodless** [comment]
> > @DarthGandalf I think that answer your question.
> 
> I see certain benefits both ways, but okay.
> 
> Tables still look broken, but that can be fixed when the translation progress reaches one.

If you mean this one for example:
image


it is the same fix like the boxes. I'll add it to the first PR that will have a table

**Comment by DarthGandalf** [comment]
For example. Even the table on the back isn't supposed to be transparent

**Comment by davidpodless** [comment]
image

fixed

File: main_he.tex
Comment (qwrtln) [inline main_he.tex:54]: Can we use something more descriptive? I understand in the context of this PR "rtl" to mean "right to left", but after a while I won't remember what it was. I suggest `textright2left`.

File: translations/setup.tex/he.po
Comment (qwrtln) [inline translations/setup.tex/he.po:129]: `xscale` is the only change here, everything else remained as it was?

**Comment by qwrtln** [comment]
> @qwrtln how do you think, the `\textenglish` wrapping should be done in the original, like what I prototyped for title page, or purely in the translation, like what's here for the back cover and setup image?

@DarthGandalf, not sure if it's still relevant, but I think both ways have their merits. If it can be easily contained within the `po` files (as I believe `\textenglish` can), then I think it should be in translations only. However, that trick with `xscale` IMO should be provided in `metadata.tex` as 1, while defined as `-1` in `main_he.tex`, so that all the circles with coordinates don't have to be copied. Those are not musts, though. I believe those approaches will make the whole thing more DRY/KISS-like.

**Comment by qwrtln** [comment]
Or, actually, no need for that separate `xscale` definition. A simple if/else trick with right2left toggle will do.

File: translations/setup.tex/he.po
Comment (davidpodless) [reply translations/setup.tex/he.po:129]: I need to actually add the translation, but as regards to the dots location - yes, that's all that was needed

File: main_he.tex
Comment (davidpodless) [reply main_he.tex:54]: I don't mind that, but I don't see the point. RTL for Right to Left is very common (for those who use right to left languages at least...)
we can also add a comment in the metadata, if that helpful

**Review by Heegu-sama — APPROVED** [review]

**Comment by DarthGandalf** [comment]
I don't think xscale is correct solution for setup image and other images where the green dots are added by tikz: the numbers and "4k -> k4" are still reversed, from what I can see. Wrapping it by \textenglish would produce correct numbers there

**Comment by davidpodless** [comment]
> I don't think xscale is correct solution for setup image and other images where the green dots are added by tikz: the numbers and "4k -> k4" are still reversed, from what I can see. Wrapping it by \textenglish would produce correct numbers there AND avoid xscale

No, I actually want it to look like k4 for now, because in the translation it will be with hebrew letters and that should looks like that.

**Comment by davidpodless** [comment]
BTW
I still can't merge
If someone can pls merge it - I'll be thankful 

image

**Comment by qwrtln** [comment]
> BTW I still can't merge If someone can pls merge it - I'll be thankful

Yes, it's me. Can you at least respond to my comments before I approve?

**Comment by davidpodless** [comment]
> > BTW I still can't merge If someone can pls merge it - I'll be thankful
> 
> Yes, it's me. Can you at least respond to my comments before I approve?

I thought I answered everything
what did I miss?

**Comment by qwrtln** [comment]
> > > BTW I still can't merge If someone can pls merge it - I'll be thankful
> > 
> > 
> > Yes, it's me. Can you at least respond to my comments before I approve?
> 
> I thought I answered everything what did I miss?

I'm on a train, so the broadband is 💩 haha. Your responses didn't load. Let me go through them.

File: main_he.tex
Comment (qwrtln) [reply main_he.tex:54]: It is obvious to you, yes, but I've never seen that in my life before. Believe me, quite a lot of people who worked on this never dealt with RTL text. Please give it a self-explanatory name, not a comment. Those extra few characters cost you nothing, but could mean the world for others.

File: translations/setup.tex/he.po
Comment (qwrtln) [reply translations/setup.tex/he.po:129]: So I suggest defining `xscale` in `main_he.tex` as `-1`, and also providing it in `metadata.tex` as `1`, so that it is `1` as default. There will be quite a number of those images, where you will have to copy the same exact text. I suggest defining it, so that you don't have to do that. But, if you insist on copy-pasting, I won't argue.

File: main_he.tex
Comment (davidpodless) [reply main_he.tex]: As I said, I really don't mind that.

Fixed.

File: translations/setup.tex/he.po
Comment (davidpodless) [reply translations/setup.tex/he.po:129]: I have no idea how to do it
My knowledge in latex is pretty basic.
it would be nicer that way you suggested, of course.

Comment (qwrtln) [reply translations/setup.tex/he.po:129]: @DarthGandalf, care to help with this?

**Review by qwrtln — APPROVED** [review]

**Comment by DarthGandalf** [comment]
@qwrtln if I understand your idea correctly, the ifelse would still need to be copied for every .po file, right? Just in the languages where "4a 4b 4c" don't need to be touched, the msgstr can stay empty, therefore simplifying the translation. 

Right?

**Comment by qwrtln** [comment]
> @qwrtln if I understand your idea correctly, the ifelse would still need to be copied for every .po file, right? Just in the languages where "4a 4b 4c" don't need to be touched, the msgstr can stay empty, therefore simplifying the translation.
> 
> Right?

Yes, exactly. It's not a must, though, it would just make the Hebrew translation a little simpler.

**Comment by davidpodless** [comment]
> > @qwrtln if I understand your idea correctly, the ifelse would still need to be copied for every .po file, right? Just in the languages where "4a 4b 4c" don't need to be touched, the msgstr can stay empty, therefore simplifying the translation.
> > Right?
> 
> Yes, exactly. It's not a must, though, it would just make the Hebrew translation a little simpler.

it can help for a future Arabic translation as well. so it IS a better solution, I just don't know how to do it

**Pull request merged**
